File: magmagc/test/com/meti/ApplicationTest.java
package com.meti;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ApplicationTest {

    public static final Path TARGET = Paths.get(".", "Index.mgs");
    public static final Path SOURCE = Paths.get(".", "Index.java");

    private static void run() throws IOException {
        if (Files.exists(SOURCE)) {
            Files.createFile(SOURCE);
        }
    }

    @Test
    void generatesNothing() throws IOException {
        run();
        assertFalse(Files.exists(TARGET));
    }

    @AfterEach
    void tearDown() throws IOException {
        Files.deleteIfExists(SOURCE);
        Files.deleteIfExists(TARGET);
    }

    @Test
    void generatesSomething() throws IOException {
        Files.createFile(SOURCE);
        run();
        assertTrue(Files.exists(TARGET));
    }
}

File: magmac/test/js/com/meti/ApplicationTest.js
import { AfterEach }import { Test }import { IOException }import { Files }import { Path }import { Paths }import { assertFalse }import { assertTrue }from org.junit.jupiter.apifrom org.junit.jupiter.apifrom java.iofrom java.nio.filefrom java.nio.filefrom java.nio.filefrom static org.junit.jupiter.api.Assertionsfrom static org.junit.jupiter.api.Assertions
function ApplicationTest(){public static final Path SOURCE = Paths.get(".", "ApplicationTest.java")
    public static final Path TARGET = Paths.get(".", "ApplicationTest.mgs")
	def generatesSomething() : Void => {
        Files.createFile(SOURCE);
        new Application(new SingleSourceSet(SOURCE), Paths.get("."), ".mgs").run();
        assertTrue(Files.exists(TARGET));
    }
	def tearDown() : Void => {
        Files.deleteIfExists(SOURCE);
        Files.deleteIfExists(TARGET);
    }
	def generatesNothing() : Void => {
        new Application(new SingleSourceSet(SOURCE), Paths.get("."), ".mgs").run();
        assertFalse(Files.exists(TARGET));
    }
	return {};
}
module.exports = {
	ApplicationTest
};